lib/logstash/codecs/identity_map_codec.rb

```diff
@@ -48,12 +48,16 @@ def initialize(listener, interval, method_symbol)
     def start
       return self if running?
       @running.make_true
-      @thread = Thread.new() do
+      @thread = Thread.start do
+        Thread.current.tap do |thread|
+          # to be able to distinguish cleaner from auto_flusher in thread dumps
+          thread.name = @method_symbol.to_s if thread.respond_to?(:name)
+        end
```

Comment:
We can leverage [`LogStash::Util.set_thread_name()`](https://github.com/elastic/logstash/blob/master/logstash-core/lib/logstash/util.rb#L27) here:

```suggestion
        # to be able to distinguish cleaner from auto_flusher in thread dumps
        LogStash::Util.set_thread_name(@method_symbol.to_s)
```

Reply:
Thanks, that wasn't the point though ... the Java name set by the JRuby is fine, since it includes the .rb path + :line. 
Setting the name at the Ruby level just adds extra information to the Java thread naming format used by the runtime.

Reply:
changed to use the LS api instead setting the Ruby name, here's how the change looks as the Java level:
previous (`thread.name = ...`):
`"Ruby-0-Thread-1@map_cleanup: .../logstash-codec-multiline/lib/logstash/codecs/identity_map_codec.rb:51"`

new thread name:
`"IdentityMapCodec#map_cleanup"`

let me know if you have ideas for a better name a sole 'cleaner' just seemed more confusing ...